python/lsst/analysis/drp/qaUtils.py

```diff
@@ -0,0 +1,225 @@
+        vals = (cat["g" + magColName].values*coeffs["HSC-G"]
+                + cat["r" + magColName].values*coeffs["HSC-R"]
+                + cat["i" + magColName].values*coeffs["HSC-I"]
+                + cat["z" + magColName].values*coeffs["HSC-Z"] + coeffs["const"])
+        cat[coeffs["name"]] = vals
+
+    return cat
+
+
+def addGalacticExtinction(cat):
+    """Adds galactic extinction for each source
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+    """
+    pass
+
+
+def addUseForQAFlag(cat):
+    """Add a flag to say if the source should be used for QA purposes
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+    """
+
+    # TODO: This needs to be checked and probably changed
+
+    use = ((np.isfinite(cat["gPsMag"])) & (np.isfinite(cat["rPsMag"]))
+           & (np.isfinite(cat["iPsMag"])) & (np.isfinite(cat["zPsMag"]))
+           & (np.isfinite(cat["zPsMag"])) & (np.isfinite(cat["gCModelMag"]))
+           & (np.isfinite(cat["rCModelMag"])) & (np.isfinite(cat["iCModelMag"]))
+           & (np.isfinite(cat["zCModelMag"])) & (np.isfinite(cat["yCModelMag"])))
+
+    for band in ["g", "r", "i", "z", "y"]:
+        usePerBand = ((cat[band + "Centroid_flag_notAtMaximum"] == 0) & (cat[band + "Shape_flag"] == 0)
+                      & (cat[band + "PsfFlux_flag"] == 0)
+                      & (cat[band + "PixelFlags_saturatedCenter"] == 0)
+                      & (cat[band + "Extendedness_flag"] == 0)
+                      & (cat[band + "Deblend_nChild"] == 0))
+        use = use & usePerBand
+    cat["useForQAFlag"] = use
+    return cat
+
+
+def addSNColumn(cat, fluxColName="PsFlux"):
+    """Add a S/N column for each band
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+    fluxColName : `str`
+        Optional, defaults to "PsFlux"
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+    """
+
+    for band in ["g", "r", "i", "z", "y"]:
+        SN = cat[band + fluxColName].values / cat[band + fluxColName + "Err"].values
+        cat[band + "Sn" + fluxColName] = SN
+
+    return cat
+
+
+def addUseForStatsColumn(cat):
+    """Add a column to indicate if a source should be used in
+       calculating statistics.
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+     """
+
+    # TODO: Is this needed? Where do the thresholds come from?
+    # Can potentially get rid of this as S/N is needed
+    # as its own column for other plots
+    useForStats = np.zeros(len(cat))
+    SN = cat["iPsFlux"].values / cat["iPsFluxErr"].values
+
+    lowSn = np.where((SN > 500))[0]
+    useForStats[lowSn] = 2
+
+    highSn = np.where((SN > 2700))[0]
+    useForStats[highSn] = 1
```

Comment:
`i` band, `500` and `2700` are all hardcoded in this function.  From the comments above it looks like maybe this really isn't being used, but just pointing out.  And again, I thought there was some action that was taking care of the useForStats selection?